### DISCUSSION TITLE
Using python types directly in SQLAlchemy 2.0 declarative mapping

### DISCUSSION THREAD

**Discussion by askmyhat** [sqlalchemy/sqlalchemy#10690] [Usage Questions] [answered]
Hi!

What is the purpose of `Mapped` type (field1, relationship1)?
Why can't we use python types directly (field2, relationship2)?

```
class Parent(Base):
    field1: Mapped[int]
    field2: int
    relationship1: Mapped[list[Child]] = relationship()
    relationship2: list[Child] = relationship()
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

Since otherwise you can't use things like `Parent.field2.in_([1,2,3])` and expect the code to pass type checking.

Note that's only one of the way to make a python class using sqlalchemy. You are free to type field however you like with other ways of mapping classes.

Replied (I guess late) also in gerrit

**Reply by askmyhat** [reply]
Thanks a lot! It should have been obvious.
